src/simulation/dynamics/prescribedLinearTranslation/prescribedLinearTranslation.h

```diff
@@ -0,0 +1,123 @@
+/*
+ ISC License
+
+ Copyright (c) 2024, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#ifndef _PRESCRIBEDTRANS_
+#define _PRESCRIBEDTRANS_
+
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/msgPayloadDefC/PrescribedTranslationMsgPayload.h"
+#include "architecture/msgPayloadDefC/LinearTranslationRigidBodyMsgPayload.h"
+#include <Eigen/Dense>
+#include <cstdint>
+
+class PrescribedLinearTranslation: public SysModel {
+public:
+    PrescribedLinearTranslation() = default;
+    ~PrescribedLinearTranslation() = default;
+
+    void Reset(uint64_t CurrentSimNanos) override;
+    void UpdateState(uint64_t CurrentSimNanos) override;
+    
+    double getcoastOptionRampDuration() const;
+
+    void setcoastOptionRampDuration(double rampDuration);
+
+    double getTransAccelMax() const;
+
+    void setTransAccelMax(double transAccelMax);
+
+    const Eigen::Vector3d &getTransHat_M() const;
+
+    void setTransHat_M(const Eigen::Vector3d &transHat_M);
+
+    const Eigen::Vector3d &getOmega_FM_F() const;
+
+    void setOmega_FM_F(const Eigen::Vector3d &omega_FM_F);
+
+    const Eigen::Vector3d &getOmegaPrime_FM_F() const;
+
+    void setOmegaPrime_FM_F(const Eigen::Vector3d &omegaPrime_FM_F);
+
+    const Eigen::Vector3d &getSigma_FM() const;
+
+    void setSigma_FM(const Eigen::Vector3d &sigma_FM);
+
+    double getTransPosInit() const;
+
+    void setTransPosInit(double transPosInit);
+
+    ReadFunctor<LinearTranslationRigidBodyMsgPayload> linearTranslationRigidBodyInMsg;    //!< Input msg for the translational reference position and velocity
+    Message<PrescribedTranslationMsgPayload> prescribedTranslationOutMsg;                 //!< Output msg for the translational body prescribed states
```

Comment:
As this module can also be used as a FSW module, not just a dynamics module, it should also have a C-wrapped output message.  See for example `fswAlgorithms/formationFlying/formationBarycenter`.  See how the C-wrapped output message `transOutMsgC` is defined in the *.h file.  Then, in the *.cpp file you have to define the `Self_Init()` method to initialize the C-wrapped output message, and when you write the message payload to the output message, you add a second line to write the same payload to the C-wrapped output message.